src/main/java/games/strategy/engine/data/UnitCollection.java

```diff
@@ -257,4 +219,37 @@ public String toString() {
   public Iterator<Unit> iterator() {
     return Collections.unmodifiableList(m_units).iterator();
   }
+
+  @Override
+  public boolean contains(Object object) {
+    return m_units.contains(object);
+  }
+
+  @Override
+  public Object[] toArray() {
+    return m_units.toArray();
+  }
+
+  @Override
+  public <T> T[] toArray(T[] array) {
+    return m_units.toArray(array);
+  }
+
+  @Override
+  public boolean remove(Object object) {
+    final boolean result = m_units.remove(object);
+    m_holder.notifyChanged();
+    return result;
+  }
+
+  @Override
+  public boolean retainAll(Collection<?> collection) {
+    return m_units.retainAll(collection);
+  }
+
+  @Override
+  public void clear() {
+    m_units.clear();
+    m_holder.notifyChanged();
+  }
```

Comment:
All of these newly-added methods have no callers.  Nor do they have any unit tests. 😞   Is it worth adding them just to have `UnitCollection` extend `Collection<Unit>`?  I can't find any code in this PR that now relies on `UnitCollection` being a `Collection<Unit>`.

I think it's fine to change the method names in this class to align with those in `Collection`.  However, I think extending `Collection<Unit>` at this time may be a YAGNI.  Is there some near-future change you're considering where this will be required?